### ISSUE TITLE
Fine-grained PATs will also work

### ISSUE THREAD

**Issue by jagthedrummer** [peter-evans/create-pull-request#2443]
One of the strikes against using a PAT was the fact that they're not scoped to a repo, but recently GitHub has introduced "Fine-grained tokens" which can be scoped to a single repo.

I just ran some tests and can confirm that a fine-grained token will do the trick as long as you give it read/write access to "Contents" and "Pull Requests" under the "Repository Permissions" section. (Note that the "Metadata" permission will automatically be set to Read-only when you select either of the previously mentioned settings.)

Just thought I'd drop a note to help anyone who might be wondering.

![Screenshot 2023-10-11 at 14-44-57 Fine-grained Personal Access Tokens](https://github.com/peter-evans/create-pull-request/assets/58702/5f83c053-0e27-4fb6-9469-2fb8d7a9aa98)

**Comment by peter-evans** [comment]
Hi @jagthedrummer 

Fine-grained PATs may work for some use cases, but not all. They are still in beta and some GitHub APIs don't support them yet.

For example, https://github.com/peter-evans/create-pull-request/issues/1791

This is why I'm not updating the docs to make it the preferred method yet.
https://github.com/peter-evans/create-pull-request/pull/1937

**Comment by rettenbs** [comment]
One thing I noticed: If the pull requests updates the workflows of the repository, the fine grain token needs the "Workflows" permission as well. Make sense to me but it took me awhile to figure it out.

**Comment by nachitox** [comment]
@jagthedrummer, can you explain how to use this generated token with the GH action?
I still don't understand `token: ${{ secrets.PAT }}` comes from.

**Comment by peter-evans** [comment]
For the upcoming [v7 release](https://github.com/peter-evans/create-pull-request/pull/3057) I've been doing lots of testing with fine-grained permissions (both fine-grained PATs and GitHub App generated tokens). I've resolved a couple of issues and I'm now more confident that this action works fine with fine-grained tokens for all the use cases my test suite covers.

The new documentation for the v7 release is now updated to cover fine-grained tokens.

This issue can remain open for now and will be closed out on release of v7.
